index.d.ts

```diff
@@ -93,8 +93,8 @@ export function combineReducers<S>(reducers: ReducersMapObject<S>): Reducer<S>;
  * transform, delay, ignore, or otherwise interpret actions or async actions
  * before passing them to the next middleware.
  */
-export interface Dispatch<S> {
-    <A extends Action>(action: A): A;
+export interface Dispatch<A extends Action = Action> {
```

Comment:
The correct typing for `redux-promise` is something like
```ts
type Dispatch = <T>(action: PromiseAction<T>) => Promise<T>;
```

Unfortunately there's no way to express this via type parameters. Currently we augment Dispatch signature in third-party middleware typings via module augmentation, and it works fine. There's a chance that with TS 2.4 we could extend Dispatch during store enhancement. I'm going to make some experiments when I have time.